modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java

```diff
@@ -125,4 +128,9 @@ public DistanceMeasure distanceMeasure() {
             .addField("centroids", centersList)
             .toString();
     }
+
+    /** {@inheritDoc} */
+    @Override public List<Object> getDependencies() {
```

Comment:
We need to start the something like "Template Trainer Design Document" based on parent class to describe the purpose of all these methods and hidden limitations of implementation like this method, what do you think, @avplatonov ?

Reply:
What do you mean? Is this a kind of example class with a lot of comments or is it a document in wiki?

Reply:
I suggest the page for Wiki in AI Wiki (you could start from the draft in the Google Doc and after review merge it into Wiki)

Reply:
Ok! I'll do it. Unfortunately, I won't have free time the next week but I'll try to prepare some draft.